tests/errors/test_dict_key1.py

```diff
@@ -0,0 +1,6 @@
+from lpython import i32
+
+def test_dict_key1():
+    my_dict: dict[list[i32], str] = {[1, 2]: "first", [3, 4]: "second"}
```

Comment:
I think we should also have such a check on the type declaration `dict[list[i32], str]`. That is, in this particular example we should throw error on the type declaration itself.

Reply:
Done.